static/app/views/performance/traces/tracesSearchBar.tsx

```diff
@@ -1,22 +1,99 @@
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
 import SearchBar from 'sentry/components/events/searchBar';
-import type {SmartSearchBarProps} from 'sentry/components/smartSearchBar';
+import {IconAdd, IconClose} from 'sentry/icons';
 import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
 import useOrganization from 'sentry/utils/useOrganization';
 
 interface TracesSearchBarProps {
-  handleSearch: SmartSearchBarProps['onSearch'];
-  query: string;
+  handleClearSearch: (index: number) => boolean;
+  handleSearch: (index: number, query: string) => void;
+  queries: string[];
 }
 
-export function TracesSearchBar({query, handleSearch}: TracesSearchBarProps) {
+const getSpanName = (index: number) => {
+  const spanNames = [t('Span A'), t('Span B'), t('Span C')];
+  return spanNames[index];
+};
+
+export function TracesSearchBar({
+  queries,
+  handleSearch,
+  handleClearSearch,
+}: TracesSearchBarProps) {
   // TODO: load tags for autocompletion
   const organization = useOrganization();
+  const canAddMoreQueries = queries.length <= 2;
+  const localQueries = queries.length ? queries : [''];
```

Comment:
This doesnt stop the user from editing the query string and injecting more queries right?

Reply:
no, basically if it's empty we pretend to have an array, otherwise we use the provided array